src/appkit.cs

```diff
@@ -26692,4 +26692,47 @@ interface NSWindowTabGroup
 		[Export ("removeWindow:")]
 		void Remove (NSWindow window);
 	}
+
+	[Mac (10,15)]
+	[BaseType (typeof(NSTouchBarItem))]
+	[DisableDefaultCtor]
+	interface NSButtonTouchBarItem
+	{
+		[Export ("initWithIdentifier:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (string identifier);
+
+		[Static]
+		[Export ("buttonTouchBarItemWithIdentifier:title:target:action:")]
+		NSButtonTouchBarItem ButtonTouchBarItemWithIdentifier (string identifier, string title, [NullAllowed] NSObject target, [NullAllowed] Selector action);
```

Comment:
`ButtonTouchBarItemWithIdentifier` sounds too much like ObjC

For .net we want 
* method names that starts with a verb (action);
* method names that do not duplicate the type name  (like `ButtonTouchBarItem`);
* method names that do not duplicate the parameter names;

The last one can be a bit confusing, e.g. `buttonTouchBarItemWithIdentifier:title:target:action:` means `buttonTouchBarItem` along with arguments
* `WithIdentifier:` (a good hint is `WIth` or `For` - they, and what's after, are generally not required)
* `title:`
* `target:
* `action:`

IOW ` {method}[{arg1}:][{arg2:}]...`, not the lack of `:` after the method name.

So a good .net method name would simply be `Create`

In other cases, like if the API was not creating instance of `NSButtonTouchBarItem` then you would need to add a description, e.g. `CreateFoo`.